video/out/opengl/video.c

```diff
@@ -3006,13 +3006,13 @@ void gl_video_render_frame(struct gl_video *p, struct vo_frame *frame,
     if (!has_frame || !mp_rect_equals(&p->dst_rect, &target_rc)) {
         struct m_color c = p->clear_color;
         float color[4] = {c.r / 255.0, c.g / 255.0, c.b / 255.0, c.a / 255.0};
-        p->ra->fns->clear(p->ra, target.tex, color, &target_rc);
+        p->ra->fns->clear(p->ra, target.tex, color);
```

Comment:
@wm4 That's what I initially had in mind as well but I'm not sure if it will work out for both vulkan and opengl. It might also result in uglier code if we try and make a no-op renderpass translate to just `glClear` on OpenGL. (Also, does all of our code handle an “empty” renderpass, i.e. with no inputs, no vertex parameters and no vertices?)

But yeah, it might be better to do this sort of “deferred application” stuff in a higher level. I just think it would result in the least complicated code to do it in ra_vk especially since vo_opengl has it map to glClear anyway. (Although doing the glClear after binding the framebuffer etc. but before the draw call might result in faster code on opengl too? Maybe that “insert shader” slow path you were talking about is for when you try clearing outside of a render pass)